app/api/route/handler/example_person_handler.py

```diff
@@ -0,0 +1,58 @@
+from datetime import date
+from typing import Optional
+from uuid import uuid4
+
+from pydantic import UUID4
+
+import api.logging
+from api.db.models.example_models import ExamplePerson, ExamplePet
+from api.util.api_context import ApiContext
+from api.util.pydantic_util import PydanticBaseModel
+
+logger = api.logging.get_logger(__name__)
+
+
+class ExamplePetParams(PydanticBaseModel):
+    example_pet_id: Optional[UUID4]
+    name: str
+    species: str
+
+
+class ExamplePersonParams(PydanticBaseModel):
+    example_person_id: Optional[UUID4]
+    first_name: str
+    middle_name: Optional[str]
+    last_name: str
+    phone_number: str
+    date_of_birth: date
+    is_real: bool
+    pets: Optional[list[ExamplePetParams]]
+
+
+class ExamplePersonResponse(ExamplePersonParams):
```

Comment:
for future discussion: i'm not sure i see much value in doing pydantic model validation on the server response.

Reply:
I definitely think the API validation as a whole could use a bit of work. I feel like more of that should be done in openapi (or equivalent). Except for potential business rules that require DB lookups or complex algorithms, should ideally not need Pydantic like this.

Reply:
But I also loathe OpenAPIs validation as it's immensely finnicky, hard to follow, and missing a lot of pretty basic things. See: how I had to structure the request/response models in it for this PR just to have the ID fields not be in the request, but be in the response

Reply:
Might be worth looking into alternatives. Openapis main benefit is the swagger page, which I'm not sure we'd get in another way.

We do already have the `CustomRequestBodyValidator` implemented, but only add one or two generic errors. Perhaps we could do something with that?

Reply:
Actually, I remember coming across it a few times, but it seems like FastAPI https://github.com/tiangolo/fastapi is "write pydantic models, generate swagger" which might be better.

Reply:
Yeah, it seems like Flask it pretty tightly coupled to openapi, you have to specify an openapi to even initialize an API. 

While there are some generators out there, they don't necessarily look easier to work with (mostly just adding annotations filled with JSON/yaml to the entry-point methods). The best chance to have the models defined in one place would be to pivot the API entirely to FastAPI which seems to have been built explicitly to work this way,  but that would probably be a bit of work as we'd need to rework all of the API code.

Reply:
(Maybe this? https://github.com/kuimono/openapi-schema-pydantic )

Reply:
I created a follow-up issue to investigate if there's a way to make the validation more straightforward (not a weird mix of openapi+pydantic): https://github.com/navapbc/template-application-flask/issues/6 along with a few potential solutions.